tree/merkle_tree.rs

```diff
@@ -0,0 +1,301 @@
+use ink_env::hash::{Blake2x256, CryptoHash};
+use ink_prelude::vec::Vec;
+use ink_storage::traits::{PackedLayout, SpreadLayout, StorageLayout};
+
+/// Merkle tree history size
+pub const ROOT_HISTORY_SIZE: u64 = 30;
+
+/// Merkle tree maximum depth
+pub const MAX_DEPTH: usize = 20;
+
+///Merkle tree with history for storing commitments in it
+#[derive(scale::Encode, scale::Decode, PackedLayout, SpreadLayout, PartialEq)]
+#[cfg_attr(feature = "std", derive(Debug, scale_info::TypeInfo, StorageLayout))]
```

Comment:
```suggestion
#[cfg_attr(feature = "std", derive(Debug, StorageLayout))]
```
I don't think we need to derive `TypeInfo` here since we will not use this MerkleTree in the contract's methods.